ci/pipeline

```diff
@@ -221,3 +236,45 @@ def jenkins(): Unit = {
   if(utils.isPullRequest) pr()
   else master()
 }
+
+/**
+ * Executes the Community Release which includes:
+ *  1. tarball with version details
+ *  2. release of docker with that version
+ *  3. git tag of version (pushed to server)
+ *
+ * @return Version and artifact description of Marathon build.
+ */
+@main
+def release(requestVersion: String, gitSha: String, latest: String = "False"): Unit = {
+
+  val releaseVersion = cleanVersion(requestVersion)
+  val tagVersion = s"v$releaseVersion"
+  println(s"Releasing version: $releaseVersion ")
+  %('git, "checkout", gitSha)
+  %('git, "tag", "-a", tagVersion, "-m", tagVersion)
+  val version = build()
+
+  uploadTarballPackagesToS3(version, s"releases/$releaseVersion")
+
+  %('docker, "push", s"mesosphere/marathon:$tagVersion")
+
+  if (latest.toUpperCase == "TRUE") {
+    %('docker, "tag", s"mesosphere/marathon:$tagVersion", "mesosphere/marathon:latest")
+    %('docker, "push", "mesosphere/marathon:latest")
+  } else {
+    println("Docker image mesosphere/marathon:latest NOT updated")
+  }
+
+  %('git, "push", "--tags")
```

Comment:
We both talked about this before, what happens if the tag already exists? Is there a way to skip the uploads? How can we make it idempotent?

Reply:
the tag early in the the process fails if the tag exists and will not reach this point.